js/utils/settingsAPIAdaptor.js

```diff
@@ -0,0 +1,201 @@
+/*
+
+  This is an adaptor between the settings model and the settings API.
+
+  === Settings API Format === 
+  
+    Current as of: https://github.com/OpenBazaar/openbazaar-go/commit/3b4d7386669b7b5a1f99e90623ecd72f3224ba28
+
+    The settings API currently returns for the models to parse in the following format:
+
+    {
+      "paymentDataInQR": false,
+      "showNotificatons": null,
+      "showNsfw": true,
+      "shippingAddresses": [],
+      "localCurrency": "USD",
+      "country": "UNITED_STATES",
+      "language": "en-US",
+      "termsAndConditions": "",
+      "refundPolicy": "",
+      "blockedNodes": [],
+      "storeModerators": [],
+      "smtpSettings": {
+        "notifications": true,
+        "serverAddress": "",
+        "username": "",
+        "password": "",
+        "senderEmail": "",
+        "recipientEmail": ""
+      }
+    }
+
+  === Settings Model toJSON Format ===
+
+    Current as of: https://github.com/tenthhuman/openbazaar-desktop/commit/deaaddfc3c1903b68100846651e98d499968c285
+
+    While the settings API currently produces a JSON object in this form:
+
+    {
+      "paymentDataInQR": false,
+      "showNotifications": true,
+      "showNsfw": true,
+      "localCurrency": "USD",
+      "country": "UNITED_STATES",
+      "language": "en-US",
+      "termsAndConditions": "",
+      "refundPolicy": "",
+      "blockedNodes": [],
+      "storeModerators": [],
+      "shippingAddresses": [],
+      "appearanceSettings": {
+        "showAdvancedVisualEffects": true,
+        "windowControlStyle": "mac"
+      },
+      "transactionSettings": {
+        "saveTransactionMetadata": true,
+        "defaultFee": "high"
+      },
+      "serverSettings": {
+        "currentFirewallType": "none"
+      },
+      "smtpIntegrationSettings": {
+        "smtpNotifications": false,
+        "smtpServerAddress": "",
+        "smtpUserName": "",
+        "smtpPassword": "",
+        "smtpFromEmail": "",
+        "smtpToEmail": ""
+      },
+      "showNotificatons": null,
+      "smtpSettings": {
+        "notifications": true,
+        "serverAddress": "",
+        "username": "",
+        "password": "",
+        "senderEmail": "",
+        "recipientEmail": ""
+      }
+    }
+
+  === What's the difference? ===
+
+  The current settings API produces an object which has a spelling error in 1 key:
+
+  'showNotificatons' works to be 'showNotifications'
+
+  Apart from that the only difference between the 
+  settings model format and the
+  settings API format is that the
+  settings model format contains 4 new keys, namely:
+
+  smtpIntegrationSettings,
+  appearanceSettings,
+  transactionSettings,
+  serverSettings
+
+  Of these 1 key, 'smtpIntegrationSettings', is substantially similar
+  to an existing key, 'smtpSettings'. 
+
+  So the proposal for the adaptor is:
+
+  - convert smtpIntegrationSettings to smtpSettings format.
+  - convert showNotifications to showNotificaton
```

Comment:
I totally agree an adapter is not what is wanted here as well. The adapter was just a way to get the functionality working and demonstrate that maybe some parts of the API could change, and that some parts of the API and requirements were not the same. I wasn't going to assume that "smtpSettings" was lore, when the spec read "SMTP Integration" settings. Seems tho that it also would work if I sought clarification from someone on the name earlier on.